src/plugin/debugger/mhpdbgc.c

```diff
@@ -2415,20 +2362,30 @@ static int bpchk(unsigned int addr)
       dpmimode = mhpdbgc.brktab[i].is_dpmi;
       trapped_bp_ = i;
       trapped_bp = -2;
+      if (mhpdbgc.brktab[i].is_one_shot) {
+        uint8_t opcode = READ_BYTE(mhpdbgc.brktab[i].brkaddr);
```

Comment:
Maybe just call the func to clear break-point?

Reply:
The existing function `mhp_clearbp()` doesn't take an index, only the addr, so it would need to iterate through again. I could make a new function that cleared by index and both could use that if necessary.

Reply:
I suspect, even just switching `mhp_clearbp()`
to use index instead of addr is possible, as there
is always index in `trapped_bp`.

Reply:
I'm just switching `mhp_clearbp()` to take an index, but in order to do so I need to use `trapped_bp` from mhpdbg.c and it's static in mhpdbgc.c. So would you prefer either making it non static, or create a new field in mhpdgbc structure?

Reply:
Probably field is better, see what suits better. :)

Reply:
Actually whilst switching `mhp_clearbp()` to take an index is great for my new stuff I'm finding `trapped_bp` is not available for bpload tidy up, primarily because bpload doesn't use `bkchk()` to test the breakpoint but does it itself. I suspect that bpload could be implemented by using the one_shot breakpoint mechanism now instead of tidying up at program exit, but I'm not looking to make this PR any bigger. So what's best, iterate in bpload tidyup to find the index, or revert to present `mhp_clearbp()` with addr and a new separate function to clear bp by index for my new stuff?